### PR TITLE
ROC-6642: Add intentionToProceedDeadline to claim

### PR DESCRIPTION
**Pull request by daveoh**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-6642

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/cmc/claimstore/utils/CaseDetailsConverter.java [hunk 1]
```diff
@@ -15,14 +19,35 @@
 
     private final CaseMapper caseMapper;
     private final JsonMapper jsonMapper;
+    private final int intentionToProceedDeadline;
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/utils/CaseDetailsConverter.java]: This definitely is not the right place for this field. I would not expect any change in this utility class for your story.

Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/utils/CaseDetailsConverter.java]: @daveoh We should not have any business logic as part of our CCD mappers, so there is no point of injecting WorkingDayIndicator in ccd-adapter. The mappers should be simply mapping to/from. 

I would expect this business logic to be part of the action/trigger to calculate this deadline set this field in `Claim` then mapper will have simple job to map this and save in CCD. When we will read this claim back from CCD, we will simply map it back from CCD to our `Claim` object.

File: src/main/java/uk/gov/hmcts/cmc/claimstore/utils/CaseDetailsConverter.java [hunk 2]
```diff
@@ -41,4 +66,14 @@ private CCDCase extractCCDCase(Map<String, Object> mapData) {
     public Map<String, Object> convertToMap(CCDCase ccdCase) {
         return (Map<String, Object>) jsonMapper.convertValue(ccdCase, Map.class);
     }
+
+    private LocalDate calculateIntentionToProceedDeadline(LocalDateTime respondedDate) {
+        LocalDate deadline = respondedDate.toLocalDate().plusDays(this.intentionToProceedDeadline);
+
+        while (!workingDayIndicator.isWorkingDay(deadline)) {
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/utils/CaseDetailsConverter.java]: FYI - To get next working day we have this method to be used `workingDayIndicator.getNextWorkingDay(...)` which has unit tests as well.

File: src/main/java/uk/gov/hmcts/cmc/claimstore/utils/CaseDetailsConverter.java (hunk 1 above)
Comment (daveoh) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/utils/CaseDetailsConverter.java]: @sabahirfan This has been discussed, but requires a definition release but we have a time constraint to get this released today. Are you happy to waive this for the short term?

Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/utils/CaseDetailsConverter.java]: @daveoh lets discuss that offline

**Pull request merged**